### DISCUSSION TITLE
SD error

### DISCUSSION THREAD

**Discussion by tack73** [sanni/cartreader#352] [Need help]
I created V3, but I can't  use it due to "SD error". I used a microSD module and 5V current is flowing there.
What should I do? Thanks.

**Comment by Androxilogin** [comment]
This has been discussed here a few times in the past. Check the SD card module itself for bridged connections. I have had them come from the factory this way. Also check over the whole reader for cold solder joints or bridged connections. Also see this thread: https://github.com/sanni/cartreader/discussions/322

**Reply by tack73** [reply]
I tried V5.5 and formatted a 16GB microSD card in FAT32, and V7.3 and formatted it in exFAT, but neither my cartreader works.
The SD card module don't have something bad.
I tried another microSD card but it don't work due to "SD error".
Thank you for your quick help.

**Reply by Androxilogin** [reply]
I know a few times the chip that has eight legs on the SD module has had the smallest bridge from the factory which threw me for a loop. You may want to check your GBA connector pins as well. This and the PIC12F629 due to having the closest connections. Use a multimeter in continuity mode.

**Reply by tack73** [reply]
I fixed PIC's connect and used another SD module.
My cartreader became to work!
Thank you for your support!!!

**Comment by Chomemel** [comment]
I post for help before I go any further.

I also have a V3 with a **SD error**. I almost soldered everything. Only the LED and the N64 cartridge slot aren't soldered yet. As for the OLED, it's not soldered yet but I use it through a bread board.
Here's what I have checked so far.
- every solder, redone some of them
- continuity test for SD module, for clock module, for resitros ans capacitors
- if there is no bridge : GBA slot and PIC
- tried 2 SD card (an old 8 Go Samsung and a recent [SanDisk SDHD 16 Go](https://fr.aliexpress.com/item/1005002552031275.html?spm=a2g0o.order_list.0.0.ea685e5bGTdHWy&gatewayAdapt=glo2fra))
- tried v7.3, 8.0, 8.3 and 8.4

I don't know what else I can check.
- I could try another microSD module but it takes forever to recieve a new one from aliexpress
- I could try to connect the arduino to my computer screen to see if the problem comes from the OLED (I don't know how to do it but I can search for how to)
- I could rerecheck my solders

Anything else? Is it important that everthing is soldered? Can the cart reader work without soldering the N64 slot and the LED or could that be the problem?

**Comment by sanni** [comment]
N64 slot and LED are not needed. 
Have you tried formatting the SD with SD Memory Card Formatter(including the CHS format size adjustment option)? https://www.sdcard.org/downloads/formatter/sd-memory-card-formatter-for-windows-download/

**Comment by Chomemel** [comment]
Hi Sanni thanks for the answer.

It didn't work unfortunatly. And I tried the two MicroSD cards.

Before I ask any more question, here is one: I noticed that Vcc of the SD module is connected to 5V of the arduino when the fisrt switch is on 5V. But when the switch is on 3V, the Vcc isn't connected to the 5V of the arduino anymore. Is it nomal?

**Comment by sanni** [comment]
That's normal. The SD modules VCC pin is always connected to the 5V rail. While the Arduino's VCC (marked with 5V on the Arduino PCB) is connected to either 5V or 3.3V depending on the switch.

If you have the Clock Gen installed try without it.
You can also try the older V2.7 Release with HW3. I have changed the SD library somewhere in between and this made some SD carts not work anymore while they did in older releases. So you would at least know that your wiring is ok if it would work with V2.7

**Comment by Chomemel** [comment]
Ok, I finally found out the answer for the SD module Vcc. When the cart reader is on, it's always connect to 5V. I was only testing the continuity with the switch on OFF.

As for the V2.7, execpt from the mushroom logo, the **SD ERROR** appeared again (with both SD cards). I also tried without the clock generator. Same error.
So what's else to check?
Is there a way to check if the SD mudule works proprely?

**Comment by Chomemel** [comment]
I won't give up.
Last thing I tried was to run the cod "SD -> card info" on my Arduino. What I got from the serial monito then was:

> Initializing SD card...initialization failed. Things to check:
> * is a card inserted?
> * is your wiring correct?
> * did you change the chipSelect pin to match your shield or module?

I checked the continuity between SD module and the arduino pins, and all of them were ok. So my last bet is SD module does not work correctly (I odered another one).
I'll also ckeck the solders on the SD module. And I'll rechecked all the solders on the main PCB one more time.

**Comment by sanni** [comment]
You can check the SD module under a magnifying glass, maybe the chip with the many tiny legs is not soldered correctly or maybe two of its pins are shorted together.

**Comment by Chomemel** [comment]
I ckecked again with the magnifying glass
- GBA slot: no bridge, all solders ok, no bridge, won't check this one anymore
- all R and C. erevything seems fine
- the sd module: no bridge, no pin unslodered. But my magnifying glass might be not stong enough to be 100% sure. And the continuity test is hard to do.

I was wondering if the problem could be the mega2560. Maybe the connection between the chip and the SD pin, but I don't really think it could be that.

**Comment by Chomemel** [comment]
Hurray !!!!!!!!!!
I found the problem !!!! Actually **_Androxilogin_** did it. A big, huge, gigantic thank to you.  Thanks to what you said on youtube, I found out MOSI and the GDD were connected on the SD module. There was a tini tiny point of solder, where I soldered the new straight pins, that touched the board (which is mostly groud everywhere). 
![IMG_20220629_222656](https://user-images.githubusercontent.com/107198559/176540008-c5ddc0d8-439e-4413-9602-346c6883d8de.jpg)
A huge thank to SANNI too. It's unbelievable the time you spend answering us. I learnt so much through your answers Do you even sleep sometimes ? ;-)
